src/components/Auth/Signup.js

```diff
@@ -0,0 +1,129 @@
+import axios from "axios";
+import React from "react";
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useAuthContex } from "../../contex/auth-contex";
+import "./auth.css";
+
+const Signup = () => {
+  const navigateHome = useNavigate();
+  const { authDispatch } = useAuthContex();
+  const [userDetail, setUserDetail] = useState({
+    name: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+  });
+  let name, value;
+  const setUserDetailHandler = (e) => {
+    name = e.target.name;
+    value = e.target.value;
+    setUserDetail({ ...userDetail, [name]: value });
```

Comment:
Small suggestion, you can directly set ```e.target.name``` in the dynamic object key

Reply:
I have done this for better readability